gen/__init__.py

```diff
@@ -401,6 +400,27 @@ def evaluate_var(name):
 
             self.calculate(sub_scope[choice], throw_on_error=False)
 
+        # Perform all multi-argument validations
+        for parameter_set, validate_fn in self._multi_arg_validate:
+            # Build up argument map for validate function. If any arguments are
+            # unset then skip this validate function.
+            kwargs = dict()
+            skip = False
+            for parameter in parameter_set:
+                if parameter not in self._arguments:
+                    skip = True
+                    break
+                kwargs[parameter] = self._arguments[parameter]
+            if skip:
+                continue
+
+            # Call the validatoin function, catching AssertionErrors and turning them into errors in
+            # the error dictionary.
+            try:
+                validate_fn(**kwargs)
+            except AssertionError as ex:
+                self._errors[parameter_set] = ex.args[0]
```

Comment:
Yes, we keep the first argument to the exception, but drop the rest. See the existing cases: https://github.com/dcos/dcos/blob/master/gen/__init__.py#L337

This always works for python's `assert` assertions and our manually raised AssertionErrors

This is basically saying "Catch all asserts" (because asserts are easier with a nicer syntax than lots of other options), and turn the message that was printed into the assert, to the message in a great big error dictionary.